CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+﻿cmake_minimum_required (VERSION 3.9)
+
+set(FFMPEG_NUGET_NAME "ffmpeg-lgpl")
+set(FFMPEG_NUGET_VERSION "6.1.0.2-preview")
+
+# Enable Hot Reload for MSVC compilers if supported.
+if(POLICY CMP0141)
+    cmake_policy(SET CMP0141 NEW)
+    set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "$<IF:$<AND:$<C_COMPILER_ID:MSVC>,$<CXX_COMPILER_ID:MSVC>>,$<$<CONFIG:Debug,RelWithDebInfo>:EditAndContinue>,$<$<CONFIG:Debug,RelWithDebInfo>:ProgramDatabase>>")
+endif()
+
+if(NOT WIN32)
+    set(CMAKE_C_COMPILER gcc CACHE STRING "C compiler" FORCE)
+    set(CMAKE_CXX_COMPILER g++ CACHE STRING "C++ compiler" FORCE)
+    set_property(TARGET ${PROJECT_NAME} PROPERTY C_STANDARD 23)
+endif(NOT WIN32)
+
+# Set up project information
+file(READ "VERSION" PROJECT_FILE_VERSION)
+project(Griffeye.VideoProcessor.Native VERSION ${PROJECT_FILE_VERSION} DESCRIPTION "Video frame extraction library using ffmpeg")
+
+add_library(${PROJECT_NAME} SHARED
+    src/libvx.c
+    src/filter.c
+    src/filtergraph.c
+    src/util.c
+)
+
+# Set version, not required but good practice. Also symlink the major version to the full version.
+set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION})
+set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+
+# Find and link to ffmpeg
+if(WIN32)
+    set_target_properties(${PROJECT_NAME} PROPERTIES VS_USER_PROPS "${CMAKE_CURRENT_SOURCE_DIR}/Griffeye.VideoProcessor.Native.deps.props")
+
+    # Mandatory Nuget dependencies
+    set_property(TARGET ${PROJECT_NAME}
+        PROPERTY CMAKE_VS_NUGET_PACKAGE_RESTORE "ON"
+    )
+    set_property(TARGET ${PROJECT_NAME}
+        PROPERTY VS_PACKAGE_REFERENCES "${FFMPEG_NUGET_NAME}_${FFMPEG_NUGET_VERSION}"
+    )
+else()
+    string(TOLOWER ${CMAKE_BUILD_TYPE} build_type)   
+
+    SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+    target_include_directories(${PROJECT_NAME} PUBLIC
+          "${CMAKE_CURRENT_SOURCE_DIR}/Packages/${FFMPEG_NUGET_NAME}/${FFMPEG_NUGET_VERSION}/build/native/Unix/include/x64"
```

Comment:
I don't like that we have hardcoded so much of the path here, it would be nice to read it from the targets file. But that can be a further improvement